x-pack/plugins/transform/public/app/hooks/use_create_transform.tsx

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { useMutation } from '@tanstack/react-query';
+
+import { i18n } from '@kbn/i18n';
+import { toMountPoint } from '@kbn/react-kibana-mount';
+
+import type {
+  PutTransformsRequestSchema,
+  PutTransformsResponseSchema,
+} from '../../../common/api_schemas/transforms';
+import { addInternalBasePath } from '../../../common/constants';
+import type { TransformId } from '../../../common/types/transform';
+import { getErrorMessage } from '../../../common/utils/errors';
+
+import { useAppDependencies, useToastNotifications } from '../app_dependencies';
+import { ToastNotificationText } from '../components';
+
+import { useRefreshTransformList } from './use_refresh_transform_list';
+
+export const useCreateTransform = (
+  transformId: TransformId,
+  transformConfig: PutTransformsRequestSchema
+) => {
+  const { http, i18n: i18nStart, theme } = useAppDependencies();
+  const refreshTransformList = useRefreshTransformList();
+  const toastNotifications = useToastNotifications();
+
+  function errorToast(error: unknown) {
+    toastNotifications.addDanger({
+      title: i18n.translate('xpack.transform.stepCreateForm.createTransformErrorMessage', {
+        defaultMessage: 'An error occurred creating the transform {transformId}:',
+        values: { transformId },
+      }),
+      text: toMountPoint(<ToastNotificationText text={getErrorMessage(error)} />, {
+        theme,
+        i18n: i18nStart,
+      }),
+    });
+  }
+
+  const mutation = useMutation({
+    mutationFn: () =>
```

Comment:
Not a blocker but I think it's slightly better to use the same pattern as useStartTransforms to pass transformId and transformConfig in the callback here. Or the other way around probably works just as well, for consistency. I think calling createTransform(undefined, {}) was a bit unclear.

Reply:
Good feedback! Originally I did it this way because we needed `transformId` in the outer scope of `mutationFn`. Had another look at the `react-query` docs and saw it's possible to also get the options of `mutationFn` from within the callbacks like `onError`. So I was able to refactor it in [1fa7a2e](https://github.com/elastic/kibana/pull/164296/commits/1fa7a2e84e38f3ce8c52a316ed8be92880a392cd).